src/main/java/life/qbic/io/commandline/DownloadDatasetCommand.java

```diff
@@ -0,0 +1,68 @@
+package life.qbic.io.commandline;
+
+import ch.ethz.sis.openbis.generic.OpenBIS;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.DataSet;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.Person;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import life.qbic.App;
+import life.qbic.model.download.OpenbisConnector;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Mixin;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+@Command(name = "list-data",
+    description = "lists datasets and their details for a given experiment code")
+public class DownloadDatasetCommand implements Runnable {
+
+  @Parameters(arity = "1", paramLabel = "experiment", description = "The code of the experiment data is attached to")
+  private String experimentCode;
+  @Option(arity = "1", paramLabel = "<space>", description = "Optional openBIS spaces to filter results", names = {"-s", "--space"})
+  private String space;
+  @Mixin
+  AuthenticationOptions auth = new AuthenticationOptions();
+
+    @Override
+    public void run() {
+      List<String> spaces = new ArrayList<>();
+      if (space != null) {
+        System.out.println("Querying experiment in space: " + space + "...");
+        spaces.add(space);
+      } else {
+        System.out.println("Querying experiment in all available spaces...");
+      }
+      OpenBIS authentication = App.loginToOpenBIS(auth.getPassword(), auth.getUser(), auth.getAS());
+      OpenbisConnector openbis = new OpenbisConnector(authentication);
+      List<DataSet> datasets = openbis.listDatasetsOfExperiment(spaces, experimentCode).stream()
+          .sorted(Comparator.comparing(
+              (DataSet d) -> d.getExperiment().getProject().getSpace().getCode())).collect(
+              Collectors.toList());
+      int i = 0;
+      for (DataSet dataSet : datasets) {
+        i++;
+        System.out.println("["+i+"]");
```

Comment:
You can be more descriptive here by just doing an iterative for loop with the iterator being named appropriately instead of "i". 
e.g. as dataSetCount